lib/swaps/Swaps.ts

```diff
@@ -293,7 +293,11 @@ class Swaps extends EventEmitter {
     // TODO: max cltv to limit routes
     // TODO: consider the time gap between taking the routes and using them.
     // TODO: multi route support (currently only 1)
-    // TODO: check to make sure we don't already have a deal for the requested r_hash
+
+    if (this.usedHashes.has(requestPacket.body!.r_hash)) {
```

Comment:
I'm working on adding integration tests for `Swaps` in https://github.com/ExchangeUnion/xud/pull/650

Added a skipped test for it: https://github.com/ExchangeUnion/xud/pull/650/files#diff-9b4e088bfea3d98a165805245f86a461R79

Reply:
@erkarl Thanks 👍